File: java/std_nat/src/fan/std/SoftRef.java
package fan.std;

import java.lang.ref.SoftReference;

import fanx.main.Sys;
import fanx.main.Type;

public class SoftRef {

	private static Type type = null;
	public Type typeof() { if (type == null) { type = Sys.findType("std::SoftRef"); } return type;  }
	
	SoftReference sref;
	
	public static SoftRef make(Object val) {
		SoftRef self = new SoftRef();
		self.sref = new SoftReference(val);
		return self;
	}
	
	public Object get() {
		return sref.get();
	}
}

File: library/std/js/SoftRef.js



fan.std.SoftRef = fan.sys.Obj.$extend(fan.sys.Obj);
fan.std.SoftRef.prototype.$ctor = function() {}
fan.std.SoftRef.prototype.$typeof = function() { return fan.std.SoftRef.$type; }

fan.std.SoftRef.make = function(val) {
	var self = new fan.std.SoftRef();
	self.m_val = val;
	return self;
}

fan.std.SoftRef.prototype.get = function() {
	return this.m_val;
}
